### PR TITLE
Bug 1188285 - Use GaiaIconRenderer for displaying siteIcons r=benfrancis

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0a5d39d0b3fe511ac29d08e7a6703dcab5fe13f6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=46eb60384284851c6d565b920c2b603de5e71229

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8f774811dbc90aa4343f1b626a91e9ee00647482

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1427a08694155914e5fe14ac58b7610bfae5fa14

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=073993c794784f4af013ba6d15a4ae761ebc166a

File: apps/system/js/app_chrome.js [hunk 1]
```diff
@@ -7,12 +7,16 @@
...
+/* global Icon */
+/* global devicePixelRatio */
 
 'use strict';
 
 (function(exports) {
   const DEFAULT_ICON_URL = '/style/chrome/images/default_icon.png';
   const PINNING_PREF = 'dev.gaia.pinning_the_web';
+  // 32px + 5px padding added by the Icon renderer
+  const ICON_SIZE = (32 + 4) / devicePixelRatio;
```
Comment (gmarty) [inline apps/system/js/app_chrome.js]: `32 + 4` or `5`? The comment is not consistent with the code.

File: apps/system/js/app_window.js [hunk 2]
```diff
@@ -2514,19 +2514,14 @@
    * @param {Object?} siteObj
    * @returns {Promise}
    */
-  AppWindow.prototype.getIconBlob = function ac_getIconBlob(origin, iconSize,
+  AppWindow.prototype.getIcon = function ac_getIcon(origin, iconSize,
     placeObj = {}, siteObj = {}) {
 
     return new Promise((resolve, reject) => {
       LazyLoader.load('/shared/js/icons_helper.js').then(() => {
-        IconsHelper.getIconBlob(origin, iconSize, placeObj, siteObj)
```
Comment (gmarty) [inline apps/system/js/app_window.js]: So, if we don't use `IconsHelper.getIconBlob` anymore, how do we manage to store the icons in the `icons` dataStore?

File: apps/system/js/app_chrome.js (hunk 1 above)
Comment (sfoster) [inline apps/system/js/app_chrome.js]: IconsHelper.getIcon also multiplies sizes by the devicePixelRatio, at https://github.com/mozilla-b2g/gaia/blob/master/shared/js/icons_helper.js#L49 so I'm not sure you need this here

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f5733209c69eaf570936908db613b6cd652ad1af

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b03759ab0309dcd0324140f140fe0570dd19ada3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=badd7155b8078e817bebe55cf13353f6a1cbf1ab

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cc0ca2af2258761598fa026965ef2fe9cb399ffa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cc0ca2af2258761598fa026965ef2fe9cb399ffa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=00d79ac771acc1f2443a38a978a4ddb7b1ed8475

File: apps/system/js/app_chrome.js [hunk 3]
```diff
@@ -1015,26 +1015,28 @@
...
       .then(iconObject => {
-        this.siteIcon.classList.toggle('small-icon', iconObject.isSmall);
-
         // We compare the original icon URL, otherwise there is a flickering
         // effect because a different object url is created each time.
-        if (this._currentIconUrl !== iconObject.originalUrl) {
-          this.siteIcon.style.backgroundImage = `url("${iconObject.url}")`;
-          this._currentIconUrl = iconObject.originalUrl;
+        if (this._currentIconUrl !== iconObject.url) {
+          this.siteIcon.style.backgroundImage = iconObject.url;
```
Comment (gmarty) [inline apps/system/js/app_chrome.js]: nit: why wrapping the url in 'url()' in app_window.js rather than here?

**Pull request merged**